test/unit/db/repos/blocks.js

```diff
@@ -0,0 +1,885 @@
+				return expect(db.any).to.be.calledOnce;
+			});
+
+			it('should return genesis block with correct values', function*() {
+				const genesisBlock = yield db.blocks.getGenesisBlock();
+
+				expect(genesisBlock).to.not.be.empty;
+				expect(genesisBlock).to.have.lengthOf(1);
+				expect(genesisBlock[0]).to.have.all.keys(
+					'id',
+					'payloadHash',
+					'blockSignature'
+				);
+
+				const payloadHash = genesisBlock[0].payloadHash.toString('hex');
+
+				return expect(payloadHash).to.be.eql(__testContext.config.nethash);
+			});
+		});
+
+		describe('getGenesisBlockId()', () => {
+			it('should use the correct SQL file with one parameters', function*() {
+				sinonSandbox.spy(db, 'query');
+				yield db.blocks.getGenesisBlockId('1111111');
+
+				expect(db.query.firstCall.args[0]).to.eql(sql.getGenesisBlockId);
+				expect(db.query.firstCall.args[1]).to.eql(['1111111']);
+				return expect(db.query).to.be.calledOnce;
+			});
+
+			it('should return block id if block exists for given id', function*() {
+				const genesisBlock = blocksFixtures.GenesisBlock();
+				const block = yield db.blocks.getGenesisBlockId(genesisBlock.id);
+
+				expect(block).to.be.not.empty;
+				expect(block).to.be.an('array');
+				expect(block).to.have.lengthOf(1);
+				return expect(block[0].id).to.be.eql(genesisBlock.id);
+			});
+
+			it('should return empty array if block does not exists for given id', function*() {
+				const block = yield db.blocks.getGenesisBlockId('111111');
+
+				expect(block).to.be.empty;
+				return expect(block).to.be.an('array');
+			});
+		});
+
+		describe('deleteBlock()', () => {
+			it('should use the correct SQL file with one parameters', function*() {
+				sinonSandbox.spy(db, 'none');
+				yield db.blocks.deleteBlock('1111111');
+
+				expect(db.none.firstCall.args[0]).to.eql(sql.deleteBlock);
+				expect(db.none.firstCall.args[1]).to.eql(['1111111']);
+				return expect(db.none).to.be.calledOnce;
+			});
+
+			it('should delete a block for given id if it exists', function*() {
+				const genesisBlock = blocksFixtures.GenesisBlock();
+				const before = yield db.one(
+					'SELECT count(*) from blocks WHERE id = $1',
+					[genesisBlock.id]
+				);
+				const status = yield db.blocks.deleteBlock(genesisBlock.id);
+				const after = yield db.one(
+					'SELECT count(*) from blocks WHERE id = $1',
+					[genesisBlock.id]
+				);
+
+				expect(before.count).to.be.equal('1');
+				expect(status).to.be.null;
+				return expect(after.count).to.be.equal('0');
+			});
+
+			it('should not throw an error if a block with given id does not exists', () => {
+				return expect(db.blocks.deleteBlock('111111')).to.be.eventually.equal(
+					null
+				);
+			});
+		});
+
+		describe('aggregateBlocksReward()', () => {
+			it('should use the correct SQL file with three parameters', function*() {
+				sinonSandbox.spy(db, 'query');
+				const params = {
+					generatorPublicKey: 'afafafafaf',
+					start: (+new Date() / 1000).toFixed(0),
```

Comment:
number of digits is optional, and default `0`, so just `.toFixed()` like everywhere in our system